driver-core/src/main/com/mongodb/internal/operation/DropCollectionOperation.java

```diff
@@ -85,52 +97,183 @@ public WriteConcern getWriteConcern() {
-                if (t != null) {
-                    errHandlingCallback.onResult(null, t);
-                } else {
-                    final SingleResultCallback<Void> releasingCallback = releasingCallback(errHandlingCallback, connection);
-                    executeCommandAsync(binding, namespace.getDatabaseName(), getCommand(connection.getDescription()),
-                            connection, writeConcernErrorWriteTransformer(), new SingleResultCallback<Void>() {
-                                @Override
-                                public void onResult(final Void result, final Throwable t) {
-                                    if (t != null && !isNamespaceError(t)) {
-                                        releasingCallback.onResult(null, t);
-                                    } else {
-                                        releasingCallback.onResult(result, null);
-                                    }
-                                }
-                            });
-                }
+        SingleResultCallback<Void> errHandlingCallback = errorHandlingCallback(callback, LOGGER);
+        getEncryptedFields((AsyncReadWriteBinding) binding, (result, t) -> {
+            if (t != null) {
+                errHandlingCallback.onResult(null, t);
+            } else {
+                withAsyncConnection(binding, (connection, t1) -> {
+                    if (t1 != null) {
+                        errHandlingCallback.onResult(null, t1);
+                    } else {
+                        new ProcessCommandsCallback(binding, connection, getCommands(result), releasingCallback(errHandlingCallback,
+                                connection))
+                                .onResult(null, null);
+                    }
+                });
             }
         });
     }
 
-    private BsonDocument getCommand(final ConnectionDescription description) {
+    /**
+     * With FLE2 dropping a collection can involve more logic and commands.
+     *
+     * <p>
+     * A call to a driver helper Collection.Drop(dropOptions) must check if the collection namespace (<databaseName>.<collectionName>)
+     * has an associated encryptedFields. Check for an associated encryptedFields from the following:
+     * <ol>
+     *     <li>The encryptedFields option passed in dropOptions.</li>
+     *     <li>The value of AutoEncryptionOpts.encryptedFieldsMap[<databaseName>.<collectionName>].</li>
+     *     <li>If AutoEncryptionOpts.encryptedFieldsMap is not null, run a listCollections command on the database databaseName with the
+     *          filter { "name": "<collectionName>" }. Check the returned options for the encryptedFields option.</li>
+     * </ol>
+     * </p>
+     * <p>
+     * If the collection namespace has an associated encryptedFields, then do the following operations.
+     * If any of the following operations error, the remaining operations are not attempted:
```

Comment:
Is "ns not found" considered an error?

Reply:
Not sure - this logic is due to change in DRIVERS-2326 so have asked there: https://github.com/mongodb/specifications/pull/1215

Reply:
Confirmed it wont be considered an error.